source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -782,6 +796,20 @@ Expected<std::vector<NamedMesh>, std::string> fromSceneObjFile( const char* data
         std::string_view line( data + newlines[li], newlines[li + 1] - newlines[li + 0] );
         currentMaterialName = line.substr( strlen( "usemtl" ), std::string_view::npos );
         boost::trim( currentMaterialName );
+        if ( mtl.has_value() )
+        {
+            auto it = mtl->find( currentMaterialName );
+            if ( it != mtl->end() )
+            {
+                if ( it->second.id )
+                    currentTextureId = it->second.id;
+                else
+                {
+                    maxTextureId++;
+                    it->second.id = currentTextureId = maxTextureId;
+                }
+            }
```

Comment:
What if the material isn't found? Will the previous material just be reused? What if `currentTextureId` remains uninitialized?